src/malli/generator.cljc

```diff
@@ -214,9 +212,16 @@
 (defmethod -schema-generator :nil [_ _] (gen/return nil))
 (defmethod -schema-generator :string [schema options] (-string-gen schema options))
 (defmethod -schema-generator :int [schema options] (gen/large-integer* (-min-max schema options)))
-(defmethod -schema-generator :double [schema options] (gen/double* (merge {:infinite? false, :NaN? false}
-                                                                          (m/properties schema options)
-                                                                          (-min-max schema options))))
+(defmethod -schema-generator :double [schema options]
+  (gen/double* (merge (let [props (m/properties schema
+                                                options)]
+                        {:infinite? (or (:gen/infinite? props)
+                                        (:infinite? props)
```

Comment:
this reserves 2 new plain keyword keys, let's just support the namespaced versions? If  `:double` uses the plain keys, we could use them here. Currently, it doesn't.  Same with `:NaN?`

Otherwise, looks good.

Reply:
Right, got slightly confused by a previous comment but your are right, this implies that the plain kws have an effect on validation as well. It might be useful some day to put that kind of restriction in `:double` (no Nan or Inf) but that's another topic.